loadfile/loadfile.go

```diff
@@ -0,0 +1,36 @@
+// Package loadfile provides functions to load files from a directory.
+package loadfile
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// LoadContext reads the contents of all the files in the given directory
+// into a map where the key is the filepath and file contents is the value.
+func LoadContext(dir string) (map[string][]byte, error) {
+	absDir, err := filepath.Abs(dir)
+	if err != nil {
+		return nil, fmt.Errorf("failed to obtain absolute path of context directory %s (%w)", dir, err)
+	}
+	result := map[string][]byte{}
+	err = filepath.Walk(absDir,
+		func(path string, i os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if i.Mode().IsRegular() {
+				fileData, err := os.ReadFile(path) //nolint:gosec
+				if err != nil {
+					return fmt.Errorf("failed to read file from context directory: %s (%w)", path, err)
+				}
+				path = strings.TrimPrefix(path, absDir)
+				path = strings.TrimPrefix(path, string([]byte{os.PathSeparator}))
```

Comment:
Calling `TrimPrefix` twice seems awkward, although it does allow working with or without a trailing separator. According to the documentation, `Abs` calls `Clean`, which will remove a trailing separator unless the full path is exactly `/`, so I think you can just trim `absDir + os.PathSeparator` unconditionally... although I suppose that's not a big deal unless this is called often enough that saving a few CPU cycles might be relevant. 😆